blockchain/block/utils.go

```diff
@@ -45,8 +46,8 @@ func calculateTransferAmount(acts []action.SealedEnvelope) *big.Int {
 func VerifyBlock(blk *Block) error {
 	if blk.Height() == 0 {
 		// verify genesis block hash
-		if blk.HashBlock() != GenesisHash() {
-			return errors.New("genesis block hash does not match")
+		if blk.Timestamp().Unix() != genesis.Timestamp() {
```

Comment:
In https://github.com/iotexproject/iotex-core/pull/2624/files#diff-b732308d6bc9e81bf4309030417cb3733b21208e5cbb4fbc2aa9a13ba60c3ffbR68, you made a change to return genesis.Hash in blockheader, then why switch to comparing timestamp here?

Reply:
since genesis.Hash() is returned for blk.Height == 0, the comparison here will always be equal.

so change to comparing timestamp, to make sure the block is actually generated by block.GenesisBlock(). Otherwise, a block with height =0 and arbitrary timestamp like
```
&Block{
		Header: Header{
			version:          version.ProtocolVersion,
			height:           0,
			timestamp:        time.Unix(any time, 0),
			prevBlockHash:    hash.ZeroHash256,
			txRoot:           hash.ZeroHash256,
			deltaStateDigest: hash.ZeroHash256,
			receiptRoot:      hash.ZeroHash256,
		},
	}
```
can pass VerifyBlock()